source/app/models/url.rb

```diff
@@ -0,0 +1,18 @@
+class Url < ActiveRecord::Base
+  validates :original, presence: true
+  validates :original, length: { minimum: 1 }
+  validates :original, format: {with: /\A((http|https):\/\/).*\Z/}
+
+
+  before_save :shortify
+
+  private
+  def shortify
+    if self.shortened == nil
+      self.click_count = 0
```

Comment:
Just as a note, you can specify a default in your migration. I don't know if you can do it with the `rails generate` cli but you can modify it as `add_column :urls, :click_count, :integer, default: 0` in the migration.